gpjax/abstractions.py

```diff
@@ -15,6 +15,15 @@
 from .utils import convert_seed
 
 
+@dataclass
```

Comment:
This is a nice idea, keeps everything tidy! 

Just a couple of questions:
- What advantages does having a dataclass have over a named Tuple? 
- Do we want to make ensure the trained params and history immutable after training has finished?

Reply:
(Sorry this is in reference to `InferenceState`).

Reply:
I'm not sure that there's a clear advantage of a dataclass over a tuple. I used a dataclass though for consistency within the rest of the package. Very happy to moved to a namedtuple though if you know of a clear advantage?

Agree on making `InferenceState` immutable - reflected in latest commit.